openml-api/src/main/java/com/feedzai/openml/data/schema/DatasetSchema.java

```diff
@@ -135,6 +136,9 @@ public FieldSchema getTargetFieldSchema() {
      * @return the list of predictive fields.
      */
     public List<FieldSchema> getPredictiveFields() {
```

Comment:
WDYT of:

```
return getTargetIndex()
                .map(targetIndex -> 
                    this.fieldSchemas
                            .stream()
                            .filter(field -> field.getFieldIndex() != targetIndex)
                            .collect(Collectors.toList()))
                .orElse(this.fieldSchemas);
```

I'm not a big fan of doing if/else on optional.isPresent

Reply:
but yes, is kinda subjective: FP vs imperative style. Feel free to keep

Reply:
I usually like to follow the following rule of thumb:
 - If a lambda doesn't fit in one line, avoid it. This helps make more readable code IMO :)